panic when GetConfigForClient returns nil

[crypto/tls Document](https://pkg.go.dev/crypto/tls#Config) says: If the returned Config is nil, the original Config will be used.

```go
http3.Server{
	Server: &http.Server{
		Addr:      ":https",
		Handler:   nil,
		TLSConfig: &tls.Config{
			GetConfigForClient: tlsConfigOverride,
		},
	},
	QuicConfig: &quic.Config{
		KeepAlive: true,
	},
}

func tlsConfigOverride(hello *tls.ClientHelloInfo) (*tls.Config, error) {
	if (...) {
		return config, nil
	}

	return nil, nil
}
```

But I got:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x148 pc=0x61a141]

goroutine 63 [running]:
crypto/tls.(*Config).Clone(0x0, 0x0)
        /usr/lib/go/src/crypto/tls/common.go:733 +0x41
github.com/lucas-clemente/quic-go/http3.(*Server).serveImpl.func2(0xc00027e3c0, 0xc0000b2240, 0xc00027e3c0, 0x0)
        /home/liming/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.19.2/http3/server.go:152 +0x93
github.com/marten-seemann/qtls-go1-15.(*Conn).readClientHello(0xc000241c00, 0x44a457, 0xc00052e8e8, 0x0)
        /home/liming/go/pkg/mod/github.com/marten-seemann/qtls-go1-15@v0.1.1/handshake_server.go:149 +0x4b1
github.com/marten-seemann/qtls-go1-15.(*Conn).serverHandshake(0xc000241c00, 0x0, 0xc00052e8a0)
        /home/liming/go/pkg/mod/github.com/marten-seemann/qtls-go1-15@v0.1.1/handshake_server.go:42 +0x56
github.com/marten-seemann/qtls-go1-15.(*Conn).Handshake(0xc000241c00, 0x0, 0x0)
        /home/liming/go/pkg/mod/github.com/marten-seemann/qtls-go1-15@v0.1.1/conn.go:1417 +0xc9
github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake.func1(0xc000263c80, 0xc000303da0, 0xc00052e900)
        /home/liming/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.19.2/internal/handshake/crypto_setup.go:297 +0x5a
created by github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake
        /home/liming/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.19.2/internal/handshake/crypto_setup.go:295 +0xa9
```

https://github.com/lucas-clemente/quic-go/blob/a76879c30599a5fe025efc5c60a9a45c9edb041c/http3/server.go#L136-L156

--- a/http3/server.go
+++ b/http3/server.go
@@ -140,18 +140,24 @@ func (s *Server) serveImpl(tlsConf *tls.Config, conn net.PacketConn) error {
 			if qconn, ok := ch.Conn.(handshake.ConnWithVersion); ok && qconn.GetQUICVersion() == quic.VersionDraft32 {
 				proto = nextProtoH3Draft32
 			}
-			conf := tlsConf
+			config := tlsConf
 			if tlsConf.GetConfigForClient != nil {
 				getConfigForClient := tlsConf.GetConfigForClient
 				var err error
-				conf, err = getConfigForClient(ch)
+				conf, err := getConfigForClient(ch)
 				if err != nil {
 					return nil, err
 				}
+				if conf != nil {
+					config = conf
+				}
 			}
-			conf = conf.Clone()
-			conf.NextProtos = []string{proto}
-			return conf, nil
+			if config == nil {
+				return nil, nil
+			}
+			config = config.Clone()
+			config.NextProtos = []string{proto}
+			return config, nil
 		},
 	}
 

--- a/http3/server_test.go
+++ b/http3/server_test.go
@@ -616,6 +616,22 @@ var _ = Describe("Server", func() {
 			Expect(conf.NextProtos).To(Equal([]string{"foo", "bar"}))
 			checkGetConfigForClientVersions(receivedConf)
 		})
+
+		It("works if GetConfigForClient returns a nil tls.Config", func() {
+			tlsConf := &tls.Config{GetConfigForClient: func(*tls.ClientHelloInfo) (*tls.Config, error) { return nil, nil }}
+
+			var receivedConf *tls.Config
+			quicListenAddr = func(addr string, conf *tls.Config, _ *quic.Config) (quic.EarlyListener, error) {
+				receivedConf = conf
+				return nil, errors.New("listen err")
+			}
+			s.TLSConfig = tlsConf
+			Expect(s.ListenAndServe()).To(HaveOccurred())
+			conf, err := receivedConf.GetConfigForClient(&tls.ClientHelloInfo{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(conf).ToNot(BeNil())
+			checkGetConfigForClientVersions(receivedConf)
+		})
 	})
 
 	It("closes gracefully", func() {

